Documentation/Articles/querying_for_data.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Querying for Data
+description: About Reading data
+keywords: read model, query, ddd, read, cqrs,  
+author: pavsaund
+---
+
+
+# Querying for Data
+
+## Introduction
+The query or read side of the CQRS pattern is all about exposing the current state of a system in a format that is ready to be consumed by interaction layers (ie: web-based user interface). The state should be exposed through models that are optimized for querying, and is usually updated by changes made to a system through events.
```

Comment:
Maybe something to emphasise that the read / query side can be whatever you want it to be: you can and will use a number of different models / technologies.

Off the top of my head:

The query or read side of the CQRS pattern is all about exposing the current state of your system in a format that is optimized for particular requirements.  Since there are competing and conflicting requirements, no single data storage technology or data format will be optimal for all scenarios.  It is therefore to be expected that in a sufficiently complex system there will be polyglot persistence and querying.  A typical example, where the main requirement is to display information to an end user, might involve persistence to a document database or relational database.  Alternatively, querying might involve populating a search index or nodes in a graph database. 

Since state is updated through event subscriptions, we can store the state in multiple forms and for multiple purposes, with a query optimized for a specific purpose.